### PR TITLE
feat(notion): fix various misc bugs

### LINKED ISSUE #6554: [Notion] ...Remove time from new date entry
### Extension

https://www.raycast.com/HenriChabrand/notion

### Description

Hi there, 
In Create Database Command, currently in date field the time is automatically added to notion, i would like to request to give an option to not include time.

Im not sure whether this is a bug or not since there's a time icon there but i cant seems to turn it off.

Thank you so much 

### Who will benefit from this feature?

_No response_

### Anything else?

_No response_

### LINKED ISSUE #8020: [Notion] Timezone and @remind support
### Extension

https://www.raycast.com/HenriChabrand/notion

### Description

When I create new items with date/time fields, the time is always three hours off even though the timezone is set on the field. Additionally, it would be great to support adding @remind.

### Steps To Reproduce

1. Using a Notion Database with timezone set on the field
2. Adding a new entry with Create a Page
3. Setting a specific time for the entry.

### Current Behaviour

The time is set to 3 hours ahead of the time entered.

### Expected Behaviour

The time should be set to the correct time.

### LINKED ISSUE #8805: [Notion] Unable to add page with "No Selection" Select
### Extension

https://www.raycast.com/notion/notion

### Description

image

### Steps To Reproduce

1. Create a page in the database.
2. The page should have an empty selection field selected, labeled "No Selection."
3. Attempt to add the page.

### Current Behaviour

An error occurs.

### Expected Behaviour

I'm able to add page to database

### PR DESCRIPTION
**Pull request by erics118**
## Description

Fixes #8020, fixed the tz issue, remind is not supported by API
Fixes #6554
Fixes #8805, a regression caused by #8199

changes:
- don't show recent and search pages twice
- use toISOString instead of custom format (for some reason the custom format function makes the date weird)
- don't store recent pages by storing the `Page` object locally. this also quite a few bugs, including:
  - page last_edited_time showing up as `now()` in many search and page details view
  - duped pages, when page titles are changed
  - probably other things too
- differentiate between a property with just a day and a property with a date and time
- support quick edit status properties
- slightly modify colors (not mentioned in changelog bc not important)

intended changes:
- ~allow move/show/hide properties in the create database page view~ (turns out there isn't a simple UX for this)
- ~allow move/show/hide properties in the database view~ (see above)

## Screencast

todo

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by erics118** [comment]
I encountered a 502 error from the Notion API, in the fetchPage function. Not sure if it was a bug with my code or with the Notion server

**Comment by erics118** [comment]
I changed the LocalStorage key from `RECENTLY_OPENED_PAGES` to `RECENT_PAGES` because it now holds a different type of data. I'm not sure if that was the best idea, or if there is something else I can do.

**Comment by thomaslombart** [comment]
> I changed the LocalStorage key from RECENTLY_OPENED_PAGES to RECENT_PAGES because it now holds a different type of data. I'm not sure if that was the best idea, or if there is something else I can do instead.

This will remove the recent pages for all users though.

**Comment by erics118** [comment]
Not sure why CI is unhappy. those files weren't things that I had changed

**Comment by erics118** [comment]
> This will remove the recent pages for all users though.

I think it's better to clear the recent pages for all users once rather than have to do a check to see if migration is necessary on every run.

**Comment by AlexIsMaking** [comment]
> While testing, I have encountered a 502 error multiple times from the Notion API, in the new `fetchPage` function. Not sure if it was a bug with my code or with the Notion server

Just to let you know, lots of Notion developers have been reporting an increase in 502 errors in the past week in the Notion developers Slack. So there's a good chance that this isn't specific to this extension.

**Comment by thomaslombart** [comment]
> I think it's better to clear the recent pages for all users once rather than have to do a check to see if migration is necessary on every run.

On the contrary, I'd say it's better not to affect the extension's UX for something technical. We can add the check for a couple of weeks/months and then remove it. By then, the frequent users will have migrated.

**Comment by erics118** [comment]
I have added a check and migration script, but I'm not sure how I can test this.

**Comment by thomaslombart** [comment]
I guess you could checkout the main branch, check out some pages, go back to your WIP branch and see if recents are still there.

**Comment by erics118** [comment]
Good idea. It seems to be working.

<details>
<summary>testing code</summary>

```diff
export function useRecentPages() {
  const { data, isLoading, mutate } = useCachedPromise(async () => {
+   await LocalStorage.removeItem("RECENT_PAGES");
    let data = await LocalStorage.getItem("RECENT_PAGES");
+   console.log("current data:", data);

    // try migrating the old recently opened pages to the new format
    if (!data || typeof data !== "string") {
      const oldData = await LocalStorage.getItem("RECENTLY_OPENED_PAGES");
+     console.log("old data:", oldData)
      // no old data either, return an empty array
      if (!oldData || typeof oldData !== "string") return [];

      const oldRecentPages = JSON.parse(oldData) as Page[];

      data = JSON.stringify(oldRecentPages.map((p) => new RecentPage(p)));

      // save the new data
      await LocalStorage.setItem("RECENT_PAGES", data);
+     console.log("new data:", data);
-     // remove the old data
-     await LocalStorage.removeItem("RECENTLY_OPENED_PAGES");
    }
```

output (relevant ids highlighted):

CleanShot 2023-09-29 at 17 03 25


</details>

**Comment by erics118** [comment]
Notion _should_ have fixed the 502 errors. I just got an email from a support thread about it.

**Comment by bkeys818** [comment]
Has every bug listed been fixed? If not, could we make a checklist?

**Comment by erics118** [comment]
No, there still are a few issues remaining. I listed all the changes/bug fixes that were made in the original message. They include a lot of small things that weren't ever reported as an issue.

Here is a list of all open and closed issues regarding notion:

<details>
<summary>open issues</summary>

- [ ] #8717 i dont use relations, so i will not address this
- [x] #8805 fixed
- [ ] #8579 auth/api issue, cant help with this
- [ ] #8596 auth/api issue, cant help with this
- [x] #8020 tz fixed, @remind unsupported
- [ ] #6812 i dont use relations, so i will not address this
- [ ] #7970 not addressed
- [ ] #8522 not addressed

</details>

<details>
<summary>closed issues</summary>

- [x] #8872 dupe, will be resolved in this PR
- [x] #8041 previously resolved
- [x] #8638 not supported by API, quicklinks PR kinda does this
- [x] #8397 not supported by API
- [x] #6414 previously resolved
- [x] #7778 auth/api issue, previously resolved
- [x] #7080 auth/api issue, previously resolved
- [x] #4158 auth/api issue, previously resolved
- [ ] #6635 unresolved
- [x] #6554 resolved by my PR
- [ ] #6460 auth/api issue
- [ ] #6002 auth/api issue
- [ ] #4496 auth/api issue
- [ ] #4923 auth/api issue
- [x] #5016 dupe
- [ ] #5110 unresolved
- [ ] #4363 unresolved
- [ ] #3759 unresolved
- [ ] #3673 unresolved

</details>

File: extensions/notion/package.json [hunk 1]
```diff
@@ -93,24 +93,27 @@
...
+    "linkedom": "^0.15.6",
     "node-fetch": "^3.3.2",
     "notion-to-md": "^3.1.1"
   },
   "devDependencies": {
     "@raycast/eslint-config": "^1.0.6",
-    "@types/node": "^20.5.8",
-    "@types/node-fetch": "^2.6.4",
-    "eslint": "^8.48.0",
-    "eslint-plugin-import": "^2.28.1",
```
Comment (thomaslombart) [inline extensions/notion/package.json:109]: Why removing the import plugin?

**Review by thomaslombart — COMMENTED** [review]
Looks good, apart from the curious ESLint plugin removal.

File: extensions/notion/package.json (hunk 1 above)
Comment (erics118) [reply extensions/notion/package.json:109]: Oops, adding it back.

**Comment by thomaslombart** [comment]
🚀

**Comment by christianmagill** [comment]
> 
> Here is a list of all open and closed issues regarding notion:
> 

Awesome stuff!

Not sure if I'm misreading this, or missing context, but there are a couple other issues in general that I'm aware of.

https://github.com/raycast/extensions/issues/8479, https://github.com/raycast/extensions/issues/8438

**Comment by erics118** [comment]
I've updated the original message. there were 3 missing issues (bc they didnt have the correct label). I intend to address all 3 of those in the future.

**Pull request merged**